# mCIRCrna
The goal is to integrate circadian transcriptomes with muscle snRNA-seq data to identify age and cell type dependent circadian gene signatures that demonstrate tissue chronicity in muscle.


## Table of Contents

- [mCIRCrna](#team-repo-template)
    - [Background](#Background)
    - [Data](#data)
    - [Method](#method)
        - Circadian Regulated Genes
        - Differential Expressions
        - Tools
    - [Results](#results) _Optional depending on project_
        - Circadian Regulated Genes
        - Differential Expressions
    - [References](#references)
    - [Team Members](#team-members)
         -  Lisa Shrestha, Van Nha Huynh, Kristen Coutinho, Russel Santos, Herminio Vazquez, Shufan Zhang

## Background

Goal is to identify circadian gene signatures that demonstrate tissue chronicity (2 fold expression changes over time) compared to age and cell type in muscle.

Goal would be to use the CircAge RNA-seq transcriptomic database, and cross reference muscle transcriptomes with snRNA-seq muscle database. Genes that show chronicity (2 fold changes over time course ZT1-24) are compared to snRNA-seq profiles from Myoatlas. Expression profiles of dysregulated chronic genes (e.g. Per2) would then be compared in muscle cell populations (MTJ, FAPs, type II myonuclei, etc.). Conversely, any very dysregulated muscle genes across aging (P21, 24 months, 30 months TA) could then be cross referenced for chronicity using CircAge.

## Data

McCarthy JJ, Andrews JL, McDearmon EL, Campbell KS, Barber BK, Miller BH, Walker JR, Hogenesch JB, Takahashi JS, Esser KA. Identification of the circadian transcriptome in adult mouse skeletal muscle. Physiol Genomics. 2007 Sep 19;31(1):86-95. doi: 10.1152/physiolgenomics.00066.2007. Epub 2007 Jun 5. PMID: 17550994; PMCID: PMC6080860.

CircAge: https://circaage.shinyapps.io/circaage/

MyoAtlas: https://research.cchmc.org/myoatlas/

## Method

### Circadian regulated genes

### Differential Expression

### Tools

## Results
:exclamation: _If your project yielded or intends to yield some novel analysis, please include them in your readme. It can be named something other than results as well._ :exclamation:

## References
    Ding, Haocheng, et al. Likelihood-based tests for detecting circadian rhythmicity and differential circadian patterns in transcriptomic applications. Briefings in Bioinformatics 22.6 (2021): bbab224.

## Team Members

- Shufan Zhang | shufan0519@gmail.com | Team Leader  
- Lisa Shrestha | Member
- Van Nha Huynh | Member
- Kristen Coutinho | Member
- Russel Santos | Member
- Herminio Vazquez | Member
