# docs

## mCIRCrna Abstract

Kristen Coutinho, Lisa Shrestha, Van Nha Huynh, Shufan Zhang, Russel Santos, Herminio Vazquez

Skeletal muscle is an endocrine organ that composes 40-60% of the adult body mass with  metabolic and overall health span implications. It is well accepted that with aging, circadian rhythm declines and is dependent on transcription reprogramming. As skeletal muscle contributes to a substantial portion of total body mass, it poses as a promising organ for drug therapeutics. However, its high cellular heterogeneity poses a challenge to safe druggable targets.
We hypothesize that important circadian rhythm markers in different cell types can be identified by cross-referencing circadian gene expressions from bulk RNA-seq to single-cell transcriptomics data obtained from mice muscle samples at similar age time points.
In this project, we first acquired published circadian gene expression data (McCarthy et.al. 2007) from mouse C57B6/J gastrocnemius muscle 7-10 weeks of age. We compared it to CircAge bulk transcriptomics data of C57B6/J-NIA mice which measured gene expression from skeletal muscle tissue collected every 4 hours for 48 hours in total darkness (DD), with the first time point being circadian time (CT) 18. We used the 6 month (young adult) and 27 month (old-adult) data.
Next, we utilized published snRNA-seq data that is represented in the graphical tool, myoatlas, from 5 month (young adult) and 24 month (old adult) mice tibia anterior (TA) samples. We integrated the data across age for differential expression singularly and also inputted our significantly changing circadian rhythm to understand the potential for cell specific expression.
We identified 32 circadian-regulated transcripts that have age- and cell-type-dependent differential expression patterns in mice skeletal muscle. Expression levels of 28 genes differ in Type IIx Myonuclei, Type IIb Myonuclei, and Satellite Cells at 5 months and 24 months. Differential expression of 2 transcripts were strictly between ages, while 4 were exclusive between cell types. 
This suggests that very few genes that are associated with circadian patterns change in expression during aging or are specific to certain cell types. 
For future directions, the list of genes from this project will be cross referenced to disease datasets to better understand skeletal muscle cell localization of circadian rhythm gene expression in the context of aging. Understanding the circadian gene expressions in skeletal muscle cell types during aging could lead to precise drug targeting therapeutics that supports entire body health.


## Data
1. [MyoAtlas](https://research.cchmc.org/myoatlas/)
2. [Circadian Transcriptome in adult mice] (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6080860/)
